src-self-hosted/Module.zig

```diff
@@ -2092,6 +2103,21 @@ fn analyzeExport(self: *Module, scope: *Scope, src: usize, symbol_name: []const
         .Fn => {},
         else => return self.fail(scope, src, "unable to export type '{}'", .{typed_value.ty}),
     }
+
+    var already_exported = false;
+    {
+        var it = self.decl_exports.iterator();
+        while (it.next()) |kv| {
+            const export_list = kv.value;
+            for (export_list) |e| {
+                if (std.mem.eql(u8, e.options.name, symbol_name)) {
+                    already_exported = true;
+                    break;
+                }
+            }
+        }
+    }
```

Comment:
Let's add a table to `Module` with symbol_name => `Export`. This loop can then become a table lookup. And finally, below, we will be able to add an error note, that says, "note: also exported here".